site/src/components/form.md

```diff
@@ -0,0 +1,195 @@
+
+`.form-addons` is added to attach controls together.
+
+```html
+<div class="field field-addons">
+	<div class="control">
+		<div class="select">
+			<select>
+				<option>$</option>
+				<option>€</option>
+			</select>
+		</div>
+	</div>
+	<div class="control">
+		<input class="input" type="text" placeholder="Transfer money" />
+	</div>
+	<div class="control">
+		<a class="button button-primary"> Transfer </a>
+	</div>
+</div>
+```
+
+`.field-horizontal` is used for displaying form label and field side by side.
+`field-body` is used on the control element container.
+
+```html
+<div class="field field-horizontal">
+	<div class="field-label field-label-sm">
+		<label class="label" for="example-small-label">Search</label>
+	</div>
+	<div class="field-body">
+		<div class="control">
+			<input
+				id="example-small-label"
+				class="input input-sm"
+				type="text"
+				placeholder="www.microsoft.com"
+			/>
+		</div>
+	</div>
+</div>
+<div class="field field-horizontal">
+	<div class="field-label field-label-lg">
+		<label class="label" for="example-large-label">Search</label>
+	</div>
+	<div class="field-body">
+		<div class="control">
+			<input
+				id="example-large-label"
+				class="input input-lg"
+				type="text"
+				placeholder="www.microsoft.com"
+			/>
+		</div>
+	</div>
+</div>
+```
+
+## Form Control
+
+Form control element are wrapped with a `.control` container.
+
+### Modifiers
+
+`.control-icons` is used when the control element contains icons.
+
+```html
+<div class="field">
+	<div class="control control-icons">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+	<div class="control control-icons margin-top-xxs">
+		<div class="select select-sm">
+			<select>
+				<option>A</option>
+				<option>B</option>
+			</select>
+			<span class="icon border" aria-hidden="true"> </span>
+		</div>
+	</div>
+</div>
+```
+
+`.control-icon-right` is used to display the icon on the right side of the element.
+
+```html
+<div class="field">
+	<div class="control control-icons control-icon-right">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+</div>
+```
+
+`.is-loading` is used to add a spinner icon.
+
+```html
+<div class="field">
+	<div class="control is-loading">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+	</div>
+</div>
+```
+
+### Required Indicator
+
+For control elements that are required for form submission, you can add a span element with the `.required-indicator` class inside the label or adjacent to the element.
+
+```html
+<div class="field">
+	<div class="control">
+		<label class="field-label label margin-bottom-xxs" for="sample-form-input-2">
+			First name
+			<span class="required-indicator" aria-hidden="true"></span>
+		</label>
+		<input class="input" type="text" placeholder="First name" id="sample-form-input-2" />
```

Comment:
The label's required indicator is aria-hidden, which might make sense assuming a required attribute is present on the input and provides the same info via assistive technology. 

This article suggests adding alt/aria-label for the required indicator asterisk. Might be worth getting the a11y team's input if we're not sure here. It could be annoyingly redundant to have the label and the input both announce "required".
https://www.deque.com/blog/anatomy-of-accessible-forms-required-form-fields/

Reply:
In the asterisk example, it seems like they just put the asterisk as a part of the label text. How about removing the aria-hidden and let it read "First name star/asterisk" + add the text explaining that the asterisk represents a required field (although may be impractical to add onto every form)? I'm thinking the most important piece is having the required attribute on the element if it's a truly required field. The asterisk would be backup. 

I can try to squeeze into Tuesday's a11y office hours to get another opinion.

Reply:
I was curious about how Fluent handles it. According to the example, it seems like they also skip aria-label when the asterisk is paired with a visible label. https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield